funcx/executors/high_throughput/worker_map.py

```diff
@@ -38,9 +51,174 @@ def remove_worker(self, worker_id):
 
         worker_type = self.worker_types[worker_id]
 
+        self.active_workers -= 1
         self.total_worker_type_counts[worker_type] -= 1
+        self.to_die_count[worker_type] -= 1
         self.total_worker_type_counts['slots'] += 1
 
+    def spin_up_workers(self, next_worker_q, address=None, debug=None, uid=None, logdir=None, worker_port=None):
+        """ Helper function to call 'remove' for appropriate workers in 'new_worker_map'.
+
+                        Parameters
+                        ----------
+                        new_worker_q : queue.Queue()
+                           Queue of worker types to be spun up next.
+                        address : str
+                            Address at which to connect to the workers.
+                        debug : bool
+                            Whether debug logging is activated.
+                        uid : str
+                            Worker ID to be assigned to worker.
+                        logdir: str
+                            Directory in which to write logs
+                        worker_port: int
+                            Port at which to connect to the workers.
+
+                        Returns
+                        ---------
+                        Total number of spun-up workers.
+                """
+        spin_ups = 0
+
+        logger.info("[SPIN UP] Next Worker Qsize: {}".format(next_worker_q.qsize()))
+        logger.info("[SPIN UP] Active Workers: {}".format(self.active_workers))
+        logger.info("[SPIN UP] Pending Workers: {}".format(self.pending_workers))
+        logger.info("[SPIN UP] Max Worker Count: {}".format(self.worker_count))
+
+        if next_worker_q.qsize() > 0 and self.active_workers + self.pending_workers < self.worker_count:
+            logger.debug("[SPIN UP] Spinning up new workers!")
+            num_slots = min(self.worker_count - self.active_workers - self.pending_workers, next_worker_q.qsize())
+            for _ in range(num_slots):
+
+                try:
+                    self.add_worker(worker_id=str(self.worker_counter),
```

Comment:
Where are we setting the mode ? Do we expect the mode info to be inferred from the container type string ?

Reply:
Went over this in person.